### ISSUE TITLE
Channel Payouts: Exploration & Low Fidelity Mockups (LoFi)

### ISSUE THREAD

**Issue by toiletgranny** [Joystream/atlas#2363]
This phase can include activities such as sketching,  prototyping, and discussing potential solutions internally, preparing for conducting usability tests (work tracked in a separate issue).

The expected outcome is a video walkthrough of happy paths for all user stories mentioned in the epic, along with links to Figma designs and prototypes.

**Comment by TCzechowski** [comment]
First tested iteration:
[Prototype](https://www.figma.com/proto/c4TB1l0dcftQNlHtE3VDEl/Payouts?page-id=831%3A77791&node-id=831%3A77792&viewport=241%2C48%2C0.36&scaling=scale-down&starting-point-node-id=832%3A87844)

[Figma link](https://www.figma.com/file/c4TB1l0dcftQNlHtE3VDEl/Payouts?node-id=831%3A77792)

**Comment by dmtrjsg** [comment]
Tomasz:

1. Revenue split is an event that is triggered when the Channel Owner has a Creator Token issued and they would like to withdraw money from the channel account to their member account, is that correct?
2. We (me & [@Kubu](https://joystreamworkspace.slack.com/team/U039UHPSYH3)) didn’t want to rename the button from Withdraw to “Split revenue” (or something like that) because it is still a withdrawal at the end of the day, and the outcome of this action is JOY tokens transferred from channel account to member account. Does that make sense (screenshots 1 and 2)?
3. Duration: Do we give channel owners the option to set a custom duration, or a couple of predefined options would be enough (screenshot 3)?
4. As I understand it, when the Channel Owner does a withdrawal, the transaction fee is deducted from the amount they are withdrawing, is that correct (screenshot 3)?
5. As I understand it, the Channel Owner can also be a token holder at the same time, so for example: I can set a revenue split to be 90% for me and 10% for the token holders, AND I can also own e.g. 50% of the tokens dedicated for token holders, so when I do the revenue split for 1000 $JOY, I get 900$JOY from those 90%, and also 50 $JOY from that 50% of 10% (so many %'s :joy:). Is that correct? Does the visualisation on screenshot 3 make sense?
6. I know that we wanted to incorporate staking of issuers own CRTs in the revenue split flow. Does the checkbox (screenshot 3) do the job here? Or the process is more complex?

Screenshot 1:
![image](https://user-images.githubusercontent.com/22870059/165395175-12953680-fd81-40e6-ba82-c485feb31981.png)

Screenshot 2:
![image](https://user-images.githubusercontent.com/22870059/165395195-4e64ebc4-cc0c-4719-bec4-e24d5244d15e.png)

Screenshot 3:
![image](https://user-images.githubusercontent.com/22870059/165395241-65033c0a-a7e4-48ec-b905-467cbc64af98.png)

**Comment by dmtrjsg** [comment]
Dima:

1. Not entirely. Revenue split is a time-bound process which needs to be initiated. Duration is set up during setting one up, so it has an expiration date, so to say it goes in cycles (on/off). Revenue split needs to be set up by channel owner to be “activated” and become “active”. During active revenue split, token holders can and must stake their coins, to partake in revenue split “round”.  This also applies to channel owner.
2. Withdrawal does make sense to keep here, but this would be a simple transaction specifying/confirming account to withdraw to.
3. Modal on screenshot2 is better suited for creating revenue split cycle and staking coins at the same time. This can happen at different times (e.g. you can create revenue split and then stake tokens). Staked tokens are locked till the end of the revenue split round.
4. Duration: I’d say anything from 1w to 2years with up to 7 options in the list. 1w; 2w; 1m; 3m; 6m; 12m; 24m. Custom cycles prob useful, but this could be a suitable to test when we get this validated.
5. It seems to me as we’d need to add staked amounts, since the proceedings depend on this. I recall we agreed to remove lists of avatars from the table view and only show a count of  members who staked their coins. Would be interesting to see % allocation (e.g. all tokens in circulation vs revenue staked). Since we ended on an open ended question if we want to move all history to a shared place of some sort, it is unclear how to best bring this view in-line with Tokens module layout patters. Last comment was to focus less on transaction history and more on fleshing out the widgets and I feel like we want to adopt similar visual language here since they are so close to each other.
6. Withdrawal and staking of personal tokens should exist in diff modals. In order to iron out payment mechanics, I feel like we’d need another call with Klaudiusz.. probably, but “not sure if we are making the right accents with the headline on the modal “to withdraw, you must split”..

**Comment by TCzechowski** [comment]
>1. Not entirely. Revenue split is a time-bound process which needs to be initiated. Duration is set up during setting one up, so it has an expiration date, so to say it goes in cycles (on/off). Revenue split needs to be set up by channel owner to be “activated” and become “active”. During active revenue split, token holders can and must stake their coins, to partake in revenue split “round”. This also applies to channel owner.

I do not understand how is that different from what I've said to be honest. How I understand the process:
-> I, as a Channel Owner (and Token Issuer) want to withdraw the money and click the Withdraw button
-> I set up and trigger revenue split as I MUST do it in order to withdraw money
-> Token holders stake their tokens during an active split period
-> Split period ends (becomes inactive)
-> My money is withdrawn and is transferred to my Member account.
Did I misunderstand something?

> Withdrawal does make sense to keep here, but this would be a simple transaction specifying/confirming account to withdraw to.

I think it depends on if I have a Creator Token issued or not? I mean, if I don't have a token issued then probably yeah, it would be a simple transaction. If I have a token issued though, then I need to trigger revenue split and I will receive my money after the split period ends. @bedeho can you confirm?

> Modal on screenshot2 is better suited for creating revenue split cycle and staking coins at the same time. This can happen at different times (e.g. you can create revenue split and then stake tokens). Staked tokens are locked till the end of the revenue split round.

I get it, and that's why we give the Channel Owner an option to stake them immediately or do it later just by unchecking the checkbox. I am referring to this part of the note you sent on our Slack channel:

> Revenue split
> Require staking from coin holders to qualify.
> Revenue split has latency and duration
> Inputs: Start date, amount, duration
**+ Allow issuer’s own CRTs to be staked in the same flow for revenue split.**
[Link to the message](https://joystreamworkspace.slack.com/archives/C039GLGPUQ7/p1650482171447449)

Did something change?

> Duration: I’d say anything from 1w to 2years with up to 7 options in the list. 1w; 2w; 1m; 3m; 6m; 12m; 24m. Custom cycles prob useful, but this could be a suitable to test when we get this validated.

So if I understand correctly, if I want to withdraw $JOY from my Channel account to my Member account and I set the duration to 24 months, I will receive my money after two years? Sounds kinda long to me tbh. Do we really want to give users option to set a revenue split period for that long?

> It seems to me as we’d need to add staked amounts, since the proceedings depend on this. I recall we agreed to remove lists of avatars from the table view and only show a count of members who staked their coins. Would be interesting to see % allocation (e.g. all tokens in circulation vs revenue staked). Since we ended on an open ended question if we want to move all history to a shared place of some sort, it is unclear how to best bring this view in-line with Tokens module layout patters. Last comment was to focus less on transaction history and more on fleshing out the widgets and I feel like we want to adopt similar visual language here since they are so close to each other.

It seems to me like this comment was referring to the table view for Creator Tokens (screenshot 2), and my question was about the "summary" on screenshot 3 (modal) 😊

> Withdrawal and staking of personal tokens should exist in diff modals. 

I agree that these two actions should, of course, have 2 separate entry points/triggers in the interface, but wouldn't it make sense to have an option to do both at the same time? What I mean by that is: If I want to withdraw $JOY from my channel account to my member account, I need to trigger a revenue split (assuming that I have a creator token issued), and in that revenue split, I need to stake my tokens. So wouldn't it make sense to add a "do everything I need to do in order to withdraw my money in just a single click" type of action there?

Ofcourse if I don't have a creator token issued, the modal with look completely different and will be stripped up from all unnecessary inputs and checkboxes. So probably just an amount input and a summary (to show the transaction fee etc).

Thoughts?

**Comment by bedeho** [comment]
> I think it depends on if I have a Creator Token issued or not? I mean, if I don't have a token issued then probably yeah, it would be a simple transaction. If I have a token issued though, then I need to trigger revenue split and I will receive my money after the split period ends. @bedeho can you confirm?

When doing a revenue split, everyone staking $CRT to get $JOY receive the $JOY right away, not at the end of the splitting period. This applies to the issuer who triggered the split as well. I do think we should allow the issuer to simultanously start the split AND stake some (or all) of the $CRT they have towards this split, so they don't need to do two transactions, as that is a bit annoying. If it was to work like this, the issuer would also instantly get their $JOY the moment they trigger the split, because would indeed then be the same moment they stake.

# Notice: I think I will not get involved in any product review here, as there are plenty of eyes on this already, I am only answering because I was tagged.

**Comment by TCzechowski** [comment]
@dmtrjsg @toiletgranny @KubaMikolajczyk @kdembler
**Video update:**
https://www.loom.com/share/f99911a26ea344d5907ae274207a47a1

**Comment by dmtrjsg** [comment]
Thanks for putting it together! 

Feedback:

1️⃣ **Welcome modal**

- Remove horisontal delimeter?
- Add “Do not show this message again” checkbox, =true by default (saved in local storage)

2️⃣ **Overview Tab**
Council payout - make message very brief and snappy “Payouts to your channel from Joystream council will appear in this widget if qualified. Learn more here>[link to Gitbook tbc]” 

Membership wallet - show current balance also in dollars (similar way we do for NFts?) Broader question is if we want to compliment tJoy balances outstanding or current with USD proxy (same as for NFts)

Total earned:

Total earned (small text in Channel balance widget - imho better to move to a different tab. Maybe add Total withdrawals widget and show these two in transactions tab?


3️⃣  **Overview empty state widgets:**

Default / Empty screen copy for Overiuew, Transactions and Council Payouts - to be re-worded in positive connotation. E.g. Transactions will appear here once xyz..

Loving the empty state widgets!!!  So second thought on empty state single placeholder on overview - since next step will potentially also show placeholder views (e.g. 3:11 on recorded video) we perhaps show empty widgets by default, and add an additional Info panel with just “ Widgets below will get populated once Payouts or Revenue is generated for the channel?


4️⃣ **Payout Modals:**
Payout modals - need to be reworked a little. Claim payout title ok

Claim Payout

Transfer funds 
From 
“Channel payout module account”
To
“Channel account”

Amount transferred
Transaction Fee

Signed with”
Member Controller account)

CTA = “Confirm”


5️⃣  Transactions widget and councils payotus widgets- thinking now

is it worth removing transaction history from Overview altogether? While simplifying channel payotus - e.g. only show chart of payouts (not hte latest one)..

For Transaction history "show more" from Channel Balance history can point to transactions tab, so we can remove it from here.. :)

 @TCzechowski @toiletgranny

**Comment by dmtrjsg** [comment]
I think we also need to address the states of interfaces for these cases originating from runtime implementation [covered here> :](https://github.com/Joystream/joystream/issues/3263)

<img width="990" alt="Screenshot 2022-05-16 at 18 03 45" src="https://user-images.githubusercontent.com/93924525/168646706-94b29859-9029-4d00-bef0-9ee962f5742a.png">

**Comment by toiletgranny** [comment]
Hey @dmtrjsg, since you asked for my take on your feedback before tomorrow's meeting, here it comes! :)

> 1️⃣ Welcome modal
> - Remove horisontal delimiter?
> - Add “Do not show this message again” checkbox, =true by default (saved in local storage)

- Agreed, though it's the kind of detail we would have discussed with Tomek when entering the HiFi phase.
- This dialog is actually meant to be displayed only once, so I don't think that's needed.

> 2️⃣ Overview Tab
> Council payout - make message very brief and snappy “Payouts to your channel from Joystream council will appear in this widget if qualified. Learn more here>[link to Gitbook tbc]”

I think it might be too early to discuss the exact wording right now (it's still LoFi, let's keep that in mind), but if we're doing this right now then: It's way more concise, I like it, but it might be also too vague. A reminder that the copy Tomek had in the designs performed very well in tests. While I agree it could use some clean-up here and there, I would try to preserve the overall sound of it. Here's my take:
| Before (As from Tomek's designs) | After |
| - | - |
| When the Joystream council notices and appreciates your hard work or wants to pay to use your content, additional JOYs to claim will appear here. | High-quality content is occasionally rewarded through JOY tokens by the Joystream council. When you receive a reward, it'll appear here for you to claim. |

> Membership wallet - show current balance also in dollars (similar way we do for NFts?) Broader question is if we want to compliment tJoy balances outstanding or current with USD proxy (same as for NFts)

I think it's a good idea if we do it consistently, that is, in the other two boxes as well. Ok? To me, this could easily replace what we currently have in the caption copy in all three boxes. "if we want to compliment tJoy balances outstanding..." — that part I believe is best discussed tomorrow in the meeting.

Also, while we're on the topic of this box, @TCzechowski, I remember that all test participants were struggling to decode the meaning of this and we talked about this; some of them even thought that the `123 021` was a wallet address. I didn't catch any recommendations for that in your video update. How about renaming the box from `Membership wallet` to `Membership balance`? This would be more aligned with the `Channel balance` we have right now in the middle, I believe. Wdyt?

> Total earned (small text in Channel balance widget - imho better to move to a different tab. Maybe add Total withdrawals widget and show these two in transactions tab?

If we're all okay with the suggestion to display dollar conversions in all three boxes, then I think this point is no longer relevant. 

> Default / Empty screen copy for Overiuew, Transactions and Council Payouts - to be re-worded in positive connotation. E.g. Transactions will appear here once xyz..

Personally, I think it's already way more _positive_ than the last version, under the condition we're looking at the same copy (it was changing throughout the video, I think the latest version was at around 5:50). I mean, sure, again, we'll def clean it up in the HiFi phase, but if you think it could be even more positive, then perhaps providing an example copy for one of those empty states would give us a better idea of what you had in mind, @dmtrjsg?

On another, more of a high-level note for entering the HiFi phase, @TCzechowski, let's keep in mind we agreed to refrain from using the word "money" when referring to payouts and JOY tokens.

> So second thought on empty state single placeholder on overview - since next step will potentially also show placeholder views (e.g. 3:11 on recorded video) we perhaps show empty widgets by default, and add an additional Info panel with just “ Widgets below will get populated once Payouts or Revenue is generated for the channel?

Yeah, I guess that from the technical standpoint there's not much sense in having two separate empty states, but then, I just really really like the simplicity the Overview tab is greeting you with in your first encounter with this part of Atlas. And, a reminder, this 2-step approach towards empty states was proven to be successful during usability tests, as it allowed participants to learn and explore the My payments tab progressively. I would vote for keeping it that way in the HiFi phase.

> Payout modals - need to be reworked a little. Claim payout title ok ...

I'm afraid I don't fully understand this part, so it's best if we discuss this during tomorrow's meeting. While I remember we agreed to add a bit more visibility, in the HiFi phase, on the source and target accounts taking part in the tokens transfer, I think we also need to remember that both the Payout and Withdraw modals performed very, very well in usability tests. I believe it was mostly thanks to their simplicity.

> is it worth removing transaction history from Overview altogether? [...] For Transaction history "show more" from Channel Balance history can point to transactions tab, so we can remove it from here.. :)

Why? Test participants actually liked having this quick overview in the main tab, if I remember correctly. Would love to know your thinking behind this suggestion. If anything, I would maybe try squeezing the "Channel balance" and "Transactions history" widgets together, but I'm not so sure.

> While simplifying channel payotus - e.g. only show chart of payouts (not hte latest one)..

Actually, we got a similar idea after usability tests, wondering why haven't we implemented that, @TCzechowski. I remember there was at least this one participant who also suggested dropping the focus on the _latest_ payout and making the widget more about all payouts in total.

**Comment by dmtrjsg** [comment]
@toiletgranny  thanks for the input! Looking forward to speaking tomorrow on this and your input is v helpful!

Rightly pointed out that some comments are for high-fi phase so thank you for consideration :) 

> Why? Test participants actually liked having this quick overview in the main tab, if I remember correctly. 
Channel balance is presented as a chart, so that implies transactions already :) 

The rest I guess is best to be discussed and happy to take your lead on most of them

**Comment by TCzechowski** [comment]
Lo-fi designs can be found here:
[Figma](https://www.figma.com/file/c4TB1l0dcftQNlHtE3VDEl/Channel-payouts?node-id=1311%3A158895)

Post-test walkthrough:
[Loom video](https://www.loom.com/share/388155b7bc83453a9d3af95f65d75724)

Comments from our last meeting to focus on during Hi-Fi phase:

- Dmitry: Make CTAs slightly bigger?
- Let's drop the middle-step empty state,
- Let's replace the wallet address with $ conversion in the top box. Let's have the wallet address maybe displayed somewhere else within that box, in a way that allows for copying.
- More snappy copy for empty states, as in let's put more focus on the action you can take to make something happen, eg. (very rough copy) "Sell your videos as NFTs and get rewards from council to see balance history."
- We need to provide a "learn more" link with more info on council payouts. Where is the best place for that?
- Let's drop the transactions history widget.
- Last council payout widget to be all about payouts history (not the last payout), also including information about payouts in which a particular channel owner did not receive any reward. To be confirmed with Klaudiusz if we'll have access to this historic data. Also, a suggestion from Dmitry to make it two separate widgets: one for all total payouts, and second one for payouts specifically concerning the channel owner.
- Top box label: "Available council payout" -> "Rewards from council"?
- Council payouts tab: let's add two boxes at the top. First one being about total payouts to my channel (?), second one for something concerning payouts and timing. Dmitry to confirm what exactly can be included in that second box.
- Transactions tab: let's add two boxes at the top. First one being about total incomes, second one for total withdrawals.
- Withdraw and claim dialogs: let's make sure the person claiming/withdrawing tokens knows if there are insuficcient account on the destination account to cover fees.
